# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmBuiltins.cpp
# Commit: caa8552f4f52
# Full Hash: caa8552f4f5298399e9e9193a623bc3007652df5
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-06-18 09:41:05
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Refactor wasm::Frame to reduce number of gnarly casts. r=lth,wingo
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79072
# ==============================================================================

diff -r 43921ad138e3 -r caa8552f4f52 js/src/wasm/WasmBuiltins.cpp
--- a/js/src/wasm/WasmBuiltins.cpp	Mon Jun 08 20:56:09 2020 +0000
+++ b/js/src/wasm/WasmBuiltins.cpp	Thu Jun 18 05:33:30 2020 +0000
@@ -300,17 +300,17 @@
   JitActivation* activation = CallingActivation();
   JSContext* cx = activation->cx();
   Frame* fp = activation->wasmExitFP();
-  Instance* instance = fp->tls->instance;
+  Instance* instance = fp->instance();
   const Code& code = instance->code();
   MOZ_ASSERT(code.metadata().debugEnabled);
 
   // The debug trap stub is the innermost frame. It's return address is the
   // actual trap site.
-  const CallSite* site = code.lookupCallSite(fp->returnAddress);
+  const CallSite* site = code.lookupCallSite(fp->returnAddress());
   MOZ_ASSERT(site);
 
   // Advance to the actual trapping frame.
-  fp = fp->callerFP;
+  fp = fp->wasmCaller();
   DebugFrame* debugFrame = DebugFrame::from(fp);
 
   if (site->kind() == CallSite::EnterFrame) {
@@ -505,7 +505,7 @@
       if (!CheckRecursionLimit(cx)) {
         return nullptr;
       }
-      if (activation->wasmExitFP()->tls->isInterrupted()) {
+      if (activation->wasmExitFP()->tls()->isInterrupted()) {
         return CheckInterrupt(cx, activation);
       }
       return ReportError(cx, JSMSG_OVER_RECURSED);